File: BikeDemoJava/app/src/main/java/com/magiclane/sdk/examples/bikedemojava/CategoryItem.java
/*
 * SPDX-FileCopyrightText: 2026 Magic Lane International B.V. <info@magiclane.com>
 * SPDX-License-Identifier: Apache-2.0
 *
 * Contact Magic Lane at <info@magiclane.com> for SDK licensing options.
 */

package com.magiclane.sdk.examples.bikedemojava;

import android.graphics.Bitmap;

// Represents a POI category chip in the horizontal bar.
public class CategoryItem {
    private final String name;
    private final Bitmap icon;
    private final int landmarkStoreId;
    private final int categoryId;

    public CategoryItem(String name, Bitmap icon, int landmarkStoreId, int categoryId) {
        this.name = name;
        this.icon = icon;
        this.landmarkStoreId = landmarkStoreId;
        this.categoryId = categoryId;
    }

    public String getName() {
        return name;
    }

    public Bitmap getIcon() {
        return icon;
    }

    public int getLandmarkStoreId() {
        return landmarkStoreId;
    }

    public int getCategoryId() {
        return categoryId;
    }
}

File: BikeDemo/app/src/main/kotlin/com/magiclane/sdk/examples/bikedemo/CategoryItem.kt
/*
 * SPDX-FileCopyrightText: 2026 Magic Lane International B.V. <info@magiclane.com>
 * SPDX-License-Identifier: Apache-2.0
 *
 * Contact Magic Lane at <info@magiclane.com> for SDK licensing options.
 */

package com.magiclane.sdk.examples.bikedemo

import android.graphics.Bitmap

// Represents a POI category chip in the horizontal bar.
data class CategoryItem(
    val name: String,
    val icon: Bitmap?,
    val landmarkStoreId: Int,
    val categoryId: Int,
)
